envoy/router/path_rewrite.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include "envoy/config/typed_config.h"
+#include "envoy/router/path_match.h"
+
+#include "source/common/common/logger.h"
+
+#include "absl/status/statusor.h"
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace Router {
+
+/**
+ * Creates the new route path based on the provided rewrite pattern.
+ * Subclassing Logger::Loggable so that implementations can log details.
+ */
+class PathRewriter : Logger::Loggable<Logger::Id::router> {
+public:
+  PathRewriter() = default;
+  virtual ~PathRewriter() = default;
+
+  /**
+   * Used to determine if the matcher policy is compatible.
+   *
+   * @param path_match_policy current path match policy for route
+   * @param active_policy true if user provided policy
+   * @return true if current path match policy is acceptable
+   */
+  virtual absl::Status isCompatibleMatchPolicy(PathMatcherSharedPtr path_match_policy,
```

Comment:
Here and elsewhere, it's not clear to me what "policy" means here (in the parameter, function and comment). Can we remove policy complete and use `isCompatiblePathMatcher(path_matcher, active_matcher)` (or some such)

Reply:
As it's an extension the provided proto can really be anything. That's the reasoning to keep this as "Policy". All we really can say is there is some criteria for custom matching or rewriting. That said, I'm ok to rename if "Policy" does not convoy that

Reply:
changed